Assets/MRTK/Services/CameraSystem/ProjectionOverride.cs

```diff
@@ -0,0 +1,114 @@
+        }
+#endif
+
+        /// <summary>
+        /// Coroutine function to set the camera matrices back to their defaults
+        /// </summary>
+        /// <returns>Coroutine enumerator</returns>
+        IEnumerator ResetViewMatricesOnFrameEnd()
+        {
+            yield return new WaitForEndOfFrame();
+            Camera cam = GetComponent<Camera>();
+            cam.ResetStereoViewMatrices();
+            cam.ResetStereoProjectionMatrices();
+        }
+
+        /// <inheritdoc />
+        void OnPreCull()
+        {
+            if (!EnableOverride)
+                return;
+
+            StartCoroutine(ResetViewMatricesOnFrameEnd());
+
+#if WINDOWS_UWP
+            IntPtr nativeStruct = UnityEngine.XR.XRDevice.GetNativePtr();
+            if (nativeStruct != IntPtr.Zero)
+            {
+                InternalHoloLensFrameStructure s = System.Runtime.InteropServices.Marshal.PtrToStructure<InternalHoloLensFrameStructure>(nativeStruct);
+
+                if (s.IHolographicFramePtr != IntPtr.Zero)
+                {
+                    var holographicCamera = (HolographicCamera)System.Runtime.InteropServices.Marshal.GetObjectForIUnknown(s.IHolographicCameraPtr);
+                    var holographicFrame = (HolographicFrame)System.Runtime.InteropServices.Marshal.GetObjectForIUnknown(s.IHolographicFramePtr);
+                    HolographicFramePrediction prediction = holographicFrame.CurrentPrediction;
+                    for (int i = 0; i < prediction.CameraPoses.Count; ++i)
+                    {
+                        HolographicCameraPose cameraPose = prediction.CameraPoses[i];
+
+                        // Default spatial locator is the HMD
+                        SpatialLocator defaultSpatialLocator = SpatialLocator.GetDefault();
+
+                        SpatialLocatorAttachedFrameOfReference attachedFrameOfReference = defaultSpatialLocator.CreateAttachedFrameOfReferenceAtCurrentHeading();
+                        SpatialCoordinateSystem attachedCoordinateFrame = attachedFrameOfReference.GetStationaryCoordinateSystemAtTimestamp(prediction.Timestamp);
+
+                        HolographicStereoTransform stereoProjection = cameraPose.ProjectionTransform;
+                        if (holographicCamera.CanOverrideViewport)
+                        {
+                            float ResolutionScale = 45.0f / 33.0f;
+
+
+                            Matrix4x4 leftProj = Camera.main.GetStereoProjectionMatrix(Camera.StereoscopicEye.Left);
+                            Matrix4x4 rightProj = Camera.main.GetStereoProjectionMatrix(Camera.StereoscopicEye.Right);
+                            leftProj.m00 *= ResolutionScale;
+                            leftProj.m11 *= ResolutionScale;
+                            rightProj.m00 *= ResolutionScale;
+                            rightProj.m11 *= ResolutionScale;
+                            Camera.main.SetStereoProjectionMatrix(Camera.StereoscopicEye.Left, leftProj);
```

Comment:
In general we use the CameraCache (because Camera.main calls I thought were just FindObjectsByTag/Name calls) - does it have to find the new main camera each time or would the camera cache suffice?

Reply:
Changed it to use CameraCache.  It just wasn't being used in the code I took from the issue